CMakeLists.txt

```
@@ -102,3 +102,16 @@ if (UNIT_TESTING)
   # Storage
   add_subdirectory(sdk/tests/storage/blobs)
 endif()
+
+# Fail geneation when setting MOCKS ON without GCC
+if(UNIT_TESTING_MOCKS)
+  if(UNIT_TESTING)
+    if(NOT CMAKE_C_COMPILER_ID MATCHES "GNU")
+      # Fail generation when asking for MOCK withoyt GCC
+      message(FATAL_ERROR "Unsupported Compiler Option.\nOption `UNIT_TESTING_MOCK` is not supported on ${CMAKE_C_COMPILER_ID}. It is only supported by GNU gcc.\nRemove option `UNIT_TESTING_MOCK` to build tests without using MOCK")
```

Isn't the option called `UNIT_TESTING_MOCKS` and not `UNIT_TESTING_MOCK`?
```suggestion
      message(FATAL_ERROR "Unsupported Compiler Option.\nOption `UNIT_TESTING_MOCKS` is not supported on ${CMAKE_C_COMPILER_ID}. It is only supported by GNU gcc.\nRemove option `UNIT_TESTING_MOCKS` to build tests without using MOCK")
```